### ISSUE TITLE
ValueError: 11.1 is not valid SemVer string on MacOS Big Sur

### ISSUE THREAD

**Issue by luc122c** [adamrehn/ue4-docker#125]
Output of the `ue4-docker info` command:

```
% ue4-docker info
Traceback (most recent call last):
  File "/usr/local/bin/ue4-docker", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/ue4docker/main.py", line 38, in main
    if platform.system() == 'Darwin' and DarwinUtils.isSupportedMacOsVersion() == False:
  File "/usr/local/lib/python3.9/site-packages/ue4docker/infrastructure/DarwinUtils.py", line 41, in isSupportedMacOsVersion
    return semver.compare(DarwinUtils.getMacOsVersion(), DarwinUtils.minimumRequiredVersion()) >= 0
  File "/usr/local/lib/python3.9/site-packages/semver.py", line 159, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/semver.py", line 856, in compare
    v1 = VersionInfo.parse(ver1)
  File "/usr/local/lib/python3.9/site-packages/semver.py", line 726, in parse
    raise ValueError("%s is not valid SemVer string" % version)
ValueError: 11.1 is not valid SemVer string
```

Additional details:

- Are you accessing the network through a proxy server? **No**

I'm trying to follow this demo: https://github.com/adamrehn/ue4-grpc-demo. I followed the guide here to get my MacBook setup: https://docs.adamrehn.com/ue4-docker/configuration/configuring-macos. I'm running Big Sur which is 11.1 but ue4-docker doesn't seem to be happy with that being a version? Any ue4-docker command seems to produce the same error as above.

![image](https://user-images.githubusercontent.com/7257092/105520212-50000c00-5cd2-11eb-9ff3-9b570e96aa8a.png)

**Comment by TBBle** [comment]
Yeah, using `semver.compare` is probably incorrect, Mac OS X is not versioned using SemVer. Basically, it's looked like SemVer up until Big Sur, which [has fooled a fair few programs parsing the version](https://stackoverflow.com/a/65402241/166389).

From that link the workaround is possibly to make it pretend to be a 10.x.x release:
```
SYSTEM_VERSION_COMPAT=1 ue4-docker info
```

**Comment by luc122c** [comment]
Thanks for the info @TBBle  Unfortunately, `SYSTEM_VERSION_COMPAT=1` returns 10.16 which still doesn't conform to SemVer. 
Interestingly, setting `SYSTEM_VERSION_COMPAT=2` sends me to the future and returns a version of 14.3!

**Comment by TBBle** [comment]
Drat. Thank you for testing that.

I guess Big Sur is OS XI 1, or OS X 10.16, or MacOS 14.3, counting the pre-OS X Mac OS releases. I wonder if SYSTEM_VERSION_COMPAT=3 will count versions forward from ProDOS?

**Comment by adamrehn** [comment]
Thanks for reporting this! I've switched from semver to the version parsing functionality from the [packaging](https://pypi.org/project/packaging/) package in commit ef0c05f, but I don't have a Mac for testing. @luc122c could you please check if it works?

**Comment by luc122c** [comment]
@adamrehn That seems to work! I haven't built anything yet but `ue4-docker info` works now! Thank you :)

<details>
lukenelson@Luke-MacBook-Pro ~ % ue4-docker info
ue4-docker version:         0.0.76 (latest available version is 0.0.76)
Operating system:           macOS 11.1 (Kernel Version 20.2.0)
Docker daemon version:      20.10.2
NVIDIA Docker supported:    No
Maximum image size:         No limit detected
Available disk space:       Unknown (typically means the Docker daemon is running in a Moby VM, e.g. Docker Desktop)
Total system memory:        16 GiB physical, 5 GiB virtual
Number of processors:       6 physical, 12 logical
</details>

**Comment by adamrehn** [comment]
Awesome! The patch is now live in ue4-docker version 0.0.77.
